search_params.py

```diff
@@ -4,24 +4,26 @@
 import logging
 import os
 import time
-import yaml
 from datetime import datetime
 from pathlib import Path
 
+import pytorch_lightning as pl
+import yaml
+from pytorch_lightning.utilities.parsing import AttributeDict
 from ray import tune
 
 from libmultilabel import data_utils
-from libmultilabel.evaluate import evaluate
+from libmultilabel.callbacks import EarlyStoppingWithCheckpoint
 from libmultilabel.model import Model
-from libmultilabel.utils import ArgDict, dump_log, init_device, set_seed
-
+from libmultilabel.utils import dump_log, init_device, set_seed
 
-logging.basicConfig(level=logging.INFO, format='%(asctime)s %(levelname)s:%(message)s')
+logging.basicConfig(level=logging.INFO,
+                    format='%(asctime)s %(levelname)s:%(message)s')
 
 
 class Trainable(tune.Trainable):
     def setup(self, config, data):
-        self.config = ArgDict(config)
+        self.config = AttributeDict(config)
```

Comment:
The second `set_seed()` cannot be ignored since we want to ensure the reproducibility. I think merge master would help?

Reply:
Merged